Unable to push form ui/cmd to Aggregate 1.7

#### Software versions 
Briefcase v1.17.x, operating system Ubuntu, MacOS and Windows, Aggregate v1.7.x, 

#### Problem description
Pushing form to Aggregate 1.7 stops on 
Start pushing form and submissions
Form doesn't exist in Aggregate
Sending form
and it is not continuing. 

#### Steps to reproduce the problem
Try to push any form to Aggregate 1.7

#### Expected behavior
User should be able to push form. 

#### Other information 
push ui/cmd works well with Aggregate 2.0 and 1.6

2019-09-05 14:09:04,606 [ForkJoinPool-8-worker-3] ERROR o.o.briefcase.reused.job.JobsRunner - Error running Job
java.io.UncheckedIOException: org.apache.http.client.ClientProtocolException
	at org.opendatakit.briefcase.reused.http.CommonsHttp.uncheckedExecute(CommonsHttp.java:136)
	at org.opendatakit.briefcase.reused.http.CommonsHttp.execute(CommonsHttp.java:87)
	at org.opendatakit.briefcase.push.aggregate.PushToAggregate.pushFormAndAttachments(PushToAggregate.java:167)
	at org.opendatakit.briefcase.push.aggregate.PushToAggregate.lambda$push$2(PushToAggregate.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.opendatakit.briefcase.push.aggregate.PushToAggregate.lambda$push$3(PushToAggregate.java:83)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenAccept$8(Job.java:134)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenSupply$9(Job.java:145)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenAccept$8(Job.java:133)
	at org.opendatakit.briefcase.reused.job.Job.lambda$thenRun$6(Job.java:109)
	at org.opendatakit.briefcase.reused.job.JobsRunner.lambda$launchAsync$1(JobsRunner.java:65)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: org.apache.http.client.ClientProtocolException: null
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:187)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at org.apache.http.client.fluent.Request.internalExecute(Request.java:173)
	at org.apache.http.client.fluent.Executor.execute(Executor.java:262)
	at org.opendatakit.briefcase.reused.http.CommonsHttp.uncheckedExecute(CommonsHttp.java:129)
	... 16 common frames omitted
Caused by: org.apache.http.client.NonRepeatableRequestException: Cannot retry request with a non-repeatable request entity.
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:226)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	... 21 common frames omitted

--- a/src/org/opendatakit/briefcase/reused/http/CommonsHttp.java
+++ b/src/org/opendatakit/briefcase/reused/http/CommonsHttp.java
@@ -16,7 +16,7 @@
 
 package org.opendatakit.briefcase.reused.http;
 
-import static org.apache.http.client.config.CookieSpecs.IGNORE_COOKIES;
+import static org.apache.http.client.config.CookieSpecs.STANDARD;
 import static org.apache.http.client.config.RequestConfig.custom;
 import static org.opendatakit.briefcase.reused.http.RequestMethod.POST;
 
@@ -33,6 +33,8 @@ import org.apache.http.entity.BasicHttpEntity;
 import org.apache.http.entity.ContentType;
 import org.apache.http.entity.mime.MultipartEntityBuilder;
 import org.apache.http.entity.mime.content.InputStreamBody;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.opendatakit.briefcase.reused.BriefcaseException;
 import org.opendatakit.briefcase.reused.http.response.Response;
@@ -40,38 +42,43 @@ import org.opendatakit.briefcase.reused.http.response.Response;
 public class CommonsHttp implements Http {
   private Executor executor;
   private final int maxConnections;
+  private final BasicCookieStore cookieStore;
 
-  private CommonsHttp(Executor executor, int maxConnections) {
+  private CommonsHttp(Executor executor, int maxConnections, BasicCookieStore cookieStore) {
     this.executor = executor;
     this.maxConnections = maxConnections;
+    this.cookieStore = cookieStore;
   }
 
   public static Http of(int maxConnections, HttpHost httpProxy) {
     if (!Http.isValidHttpConnections(maxConnections))
       throw new BriefcaseException("Invalid maximum simultaneous HTTP connections " + maxConnections + ". Try a value between " + MIN_HTTP_CONNECTIONS + " and " + MAX_HTTP_CONNECTIONS);
-    return new CommonsHttp(Executor.newInstance(
-        getBaseBuilder(maxConnections).setProxy(httpProxy).build()
-    ), maxConnections);
+    BasicCookieStore cookieStore = new BasicCookieStore();
+    return new CommonsHttp(Executor.newInstance(getBaseBuilder(maxConnections, cookieStore).setProxy(httpProxy).build()), maxConnections, cookieStore);
   }
 
   public static Http of(int maxConnections) {
     if (!Http.isValidHttpConnections(maxConnections))
       throw new BriefcaseException("Invalid maximum simultaneous HTTP connections " + maxConnections + ". Try a value between " + MIN_HTTP_CONNECTIONS + " and " + MAX_HTTP_CONNECTIONS);
-    return new CommonsHttp(Executor.newInstance(
-        getBaseBuilder(maxConnections).build()
-    ), maxConnections);
+    BasicCookieStore cookieStore = new BasicCookieStore();
+    HttpClientBuilder baseBuilder = getBaseBuilder(maxConnections, cookieStore);
+
+    CloseableHttpClient build = baseBuilder.build();
+
+    return new CommonsHttp(Executor.newInstance(build), maxConnections, cookieStore);
   }
 
-  private static HttpClientBuilder getBaseBuilder(int maxConnections) {
+  private static HttpClientBuilder getBaseBuilder(int maxConnections, BasicCookieStore cookieStore) {
     return HttpClientBuilder
         .create()
+        .setDefaultCookieStore(cookieStore)
         .setMaxConnPerRoute(maxConnections)
         .setMaxConnTotal(maxConnections)
         .setDefaultRequestConfig(custom()
             .setConnectionRequestTimeout(0)
             .setSocketTimeout(0)
             .setConnectTimeout(0)
-            .setCookieSpec(IGNORE_COOKIES)
+            .setCookieSpec(STANDARD)
             .build());
   }
 
@@ -89,15 +96,18 @@ public class CommonsHttp implements Http {
 
   @Override
   public void setProxy(HttpHost proxy) {
-    executor = Executor.newInstance(getBaseBuilder(maxConnections).setProxy(proxy).build());
+    executor = Executor.newInstance(getBaseBuilder(maxConnections, new BasicCookieStore()).setProxy(proxy).build());
   }
 
   @Override
   public void unsetProxy() {
-    executor = Executor.newInstance(getBaseBuilder(maxConnections).build());
+    executor = Executor.newInstance(getBaseBuilder(maxConnections, new BasicCookieStore()).build());
   }
 
   private <T> Response<T> uncheckedExecute(Request<T> request, Executor executor) {
+    if (request.ignoreCookies())
+      cookieStore.clear();
+
     // Get an Apache Commons HTTPClient request and set some reasonable timeouts
     org.apache.http.client.fluent.Request commonsRequest = getCommonsRequest(request);
 

--- a/src/org/opendatakit/briefcase/reused/http/Request.java
+++ b/src/org/opendatakit/briefcase/reused/http/Request.java
@@ -46,8 +46,9 @@ public class Request<T> {
   final Map<String, String> headers;
   private final Optional<InputStream> body;
   final List<MultipartMessage> multipartMessages;
+  private final boolean ignoreCookies;
 
-  Request(RequestMethod method, URL url, Optional<Credentials> credentials, Function<InputStream, T> responseMapper, Map<String, String> headers, Optional<InputStream> body, List<MultipartMessage> multipartMessages) {
+  Request(RequestMethod method, URL url, Optional<Credentials> credentials, Function<InputStream, T> responseMapper, Map<String, String> headers, Optional<InputStream> body, List<MultipartMessage> multipartMessages, boolean ignoreCookies) {
     this.method = method;
     this.url = url;
     this.credentials = credentials;
@@ -55,6 +56,7 @@ public class Request<T> {
     this.headers = headers;
     this.body = body;
     this.multipartMessages = multipartMessages;
+    this.ignoreCookies = ignoreCookies;
   }
 
   public T map(InputStream responseBody) {
@@ -91,7 +93,7 @@ public class Request<T> {
    * Returns a RequestBuilder that would produce this instance when built.
    */
   public RequestBuilder<T> builder() {
-    return new RequestBuilder<>(method, url, responseMapper, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, url, responseMapper, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   @Override
@@ -120,5 +122,9 @@ public class Request<T> {
 
   public InputStream getBody() {
     return body.orElseThrow(BriefcaseException::new);
+  }
+
+  public boolean ignoreCookies() {
+    return ignoreCookies;
   }
 }

--- a/src/org/opendatakit/briefcase/reused/http/RequestBuilder.java
+++ b/src/org/opendatakit/briefcase/reused/http/RequestBuilder.java
@@ -71,8 +71,9 @@ public class RequestBuilder<T> {
   private final Map<String, String> headers;
   private final Optional<InputStream> body;
   private final List<MultipartMessage> multipartMessages;
+  private final boolean ignoreCookies;
 
-  RequestBuilder(RequestMethod method, URL baseUrl, Function<InputStream, T> responseMapper, Optional<Credentials> credentials, Map<String, String> headers, Optional<InputStream> body, List<MultipartMessage> multipartMessages) {
+  RequestBuilder(RequestMethod method, URL baseUrl, Function<InputStream, T> responseMapper, Optional<Credentials> credentials, Map<String, String> headers, Optional<InputStream> body, List<MultipartMessage> multipartMessages, boolean ignoreCookies) {
     this.method = method;
     this.baseUrl = baseUrl;
     this.credentials = credentials;
@@ -80,22 +81,23 @@ public class RequestBuilder<T> {
     this.headers = headers;
     this.body = body;
     this.multipartMessages = multipartMessages;
+    this.ignoreCookies = ignoreCookies;
   }
 
   public static RequestBuilder<InputStream> get(String baseUrl) {
-    return new RequestBuilder<>(GET, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList());
+    return new RequestBuilder<>(GET, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList(), false);
   }
 
   public static RequestBuilder<InputStream> get(URL baseUrl) {
-    return new RequestBuilder<>(GET, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList());
+    return new RequestBuilder<>(GET, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList(), false);
   }
 
   public static RequestBuilder<InputStream> post(URL baseUrl) {
-    return new RequestBuilder<>(POST, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList());
+    return new RequestBuilder<>(POST, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList(), false);
   }
 
   public static RequestBuilder<InputStream> head(URL baseUrl) {
-    return new RequestBuilder<>(HEAD, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList());
+    return new RequestBuilder<>(HEAD, url(stripTrailingSlash(baseUrl)), Function.identity(), empty(), new HashMap<>(), empty(), emptyList(), false);
   }
 
   private static String readString(InputStream in) {
@@ -176,46 +178,46 @@ public class RequestBuilder<T> {
   }
 
   public Request<T> build() {
-    return new Request<>(method, baseUrl, credentials, responseMapper, headers, body, multipartMessages);
+    return new Request<>(method, baseUrl, credentials, responseMapper, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<String> asText() {
-    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readString, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readString, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<XmlElement> asXmlElement() {
-    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readXmlElement, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readXmlElement, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<Map<String, Object>> asJsonMap() {
-    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readJsonMap, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, RequestBuilder::readJsonMap, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<List<Map>> asJsonList() {
-    return new RequestBuilder<>(method, baseUrl, RequestBuilder.readJsonList(Map.class), credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, RequestBuilder.readJsonList(Map.class), credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public <U> RequestBuilder<List<U>> asJsonList(Class<U> mappingClass) {
-    return new RequestBuilder<>(method, baseUrl, RequestBuilder.readJsonList(mappingClass), credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, RequestBuilder.readJsonList(mappingClass), credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<Void> downloadTo(Path target) {
     return new RequestBuilder<>(method, baseUrl, in -> {
       copy(in, target, REPLACE_EXISTING);
       return null;
-    }, credentials, headers, body, multipartMessages);
+    }, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public <U> RequestBuilder<U> withResponseMapper(Function<T, U> mapper) {
-    return new RequestBuilder<>(method, baseUrl, responseMapper.andThen(mapper), credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper.andThen(mapper), credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withCredentials(Optional<Credentials> credentials) {
-    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withCredentials(Credentials credentials) {
-    return new RequestBuilder<>(method, baseUrl, responseMapper, Optional.of(credentials), headers, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, Optional.of(credentials), headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withPath(String path) {
@@ -223,7 +225,7 @@ public class RequestBuilder<T> {
     int endOffset = path.endsWith("/") ? 1 : 0;
     String cleanPath = path.substring(startOffset, path.length() - endOffset);
     URL newBaseUrl = url(baseUrl + "/" + cleanPath);
-    return new RequestBuilder<>(method, newBaseUrl, responseMapper, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, newBaseUrl, responseMapper, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   @SafeVarargs
@@ -232,27 +234,31 @@ public class RequestBuilder<T> {
       throw new BriefcaseException("Can't apply withQuery() twice");
     String queryString = Stream.of(keyValues).map(p -> p.getLeft() + "=" + urlEncode(p.getRight())).collect(joining("&"));
     URL newBaseUrl = url(baseUrl.toString() + "?" + queryString);
-    return new RequestBuilder<>(method, newBaseUrl, responseMapper, credentials, headers, body, multipartMessages);
+    return new RequestBuilder<>(method, newBaseUrl, responseMapper, credentials, headers, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withBody(String bodyContents) {
     Optional<InputStream> newBody = Optional.of(new ByteArrayInputStream(bodyContents.getBytes(UTF_8)));
-    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, newBody, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, newBody, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withBody(InputStream body) {
-    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, Optional.of(body), multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, Optional.of(body), multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withHeader(String name, String value) {
     Map<String, String> newHeaders = new HashMap<>(headers);
     newHeaders.put(name, value);
-    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, newHeaders, body, multipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, newHeaders, body, multipartMessages, ignoreCookies);
   }
 
   public RequestBuilder<T> withMultipartMessage(String name, String contentType, String attachmentName, InputStream messageBody) {
     List<MultipartMessage> newMultipartMessages = new ArrayList<>(multipartMessages);
     newMultipartMessages.add(new MultipartMessage(name, contentType, attachmentName, messageBody));
-    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, body, newMultipartMessages);
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, body, newMultipartMessages, ignoreCookies);
+  }
+
+  public RequestBuilder<T> withIgnoreCookies() {
+    return new RequestBuilder<>(method, baseUrl, responseMapper, credentials, headers, body, multipartMessages, true);
   }
 }

--- a/src/org/opendatakit/briefcase/reused/transfer/CentralServer.java
+++ b/src/org/opendatakit/briefcase/reused/transfer/CentralServer.java
@@ -73,6 +73,7 @@ public class CentralServer implements RemoteServer {
   public Request<String> getProjectTestRequest(String token) {
     return RequestBuilder.get(baseUrl)
         .asText()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId)
         .withHeader("Authorization", "Bearer " + token)
         .withResponseMapper(json -> json)
@@ -82,6 +83,7 @@ public class CentralServer implements RemoteServer {
   public Request<String> getSessionTokenRequest() {
     return RequestBuilder.post(baseUrl)
         .asJsonMap()
+        .withIgnoreCookies()
         .withPath("/v1/sessions")
         .withHeader("Content-Type", "application/json")
         .withBody(buildSessionPayload(credentials))
@@ -99,6 +101,7 @@ public class CentralServer implements RemoteServer {
 
   public Request<Void> getDownloadFormRequest(String formId, Path target, String token) {
     return RequestBuilder.get(baseUrl)
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + ".xml")
         .withHeader("Authorization", "Bearer " + token)
         .downloadTo(target)
@@ -108,6 +111,7 @@ public class CentralServer implements RemoteServer {
   public Request<List<CentralAttachment>> getFormAttachmentListRequest(String formId, String token) {
     return RequestBuilder.get(baseUrl)
         .asJsonList(CentralAttachment.class)
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/attachments")
         .withHeader("Authorization", "Bearer " + token)
         .build();
@@ -115,6 +119,7 @@ public class CentralServer implements RemoteServer {
 
   public Request<Void> getDownloadFormAttachmentRequest(String formId, CentralAttachment attachment, Path target, String token) {
     return RequestBuilder.get(baseUrl)
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/attachments/" + attachment.getName())
         .withHeader("Authorization", "Bearer " + token)
         .downloadTo(target)
@@ -124,6 +129,7 @@ public class CentralServer implements RemoteServer {
   public Request<List<String>> getInstanceIdListRequest(String formId, String token) {
     return RequestBuilder.get(baseUrl)
         .asJsonList()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions")
         .withHeader("Authorization", "Bearer " + token)
         .withResponseMapper(json -> json.stream().map(o -> (String) o.get("instanceId")).collect(toList()))
@@ -132,6 +138,7 @@ public class CentralServer implements RemoteServer {
 
   public Request<Void> getDownloadSubmissionRequest(String formId, String instanceId, Path target, String token) {
     return RequestBuilder.get(baseUrl)
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions/" + instanceId + ".xml")
         .withHeader("Authorization", "Bearer " + token)
         .downloadTo(target)
@@ -140,6 +147,7 @@ public class CentralServer implements RemoteServer {
 
   public Request<List<CentralAttachment>> getSubmissionAttachmentListRequest(String formId, String instanceId, String token) {
     return RequestBuilder.get(baseUrl)
+        .withIgnoreCookies()
         .asJsonList(CentralAttachment.class)
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions/" + instanceId + "/attachments")
         .withHeader("Authorization", "Bearer " + token)
@@ -148,6 +156,7 @@ public class CentralServer implements RemoteServer {
 
   public Request<Void> getDownloadSubmissionAttachmentRequest(String formId, String instanceId, CentralAttachment attachment, Path target, String token) {
     return RequestBuilder.get(baseUrl)
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions/" + instanceId + "/attachments/" + attachment.getName())
         .withHeader("Authorization", "Bearer " + token)
         .downloadTo(target)
@@ -163,6 +172,7 @@ public class CentralServer implements RemoteServer {
   public Request<Map<String, Object>> getPushFormRequest(Path formFile, String token) {
     return RequestBuilder.post(baseUrl)
         .asJsonMap()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms")
         .withHeader("Authorization", "Bearer " + token)
         .withHeader("Content-Type", "application/xml")
@@ -173,6 +183,7 @@ public class CentralServer implements RemoteServer {
   public Request<Map<String, Object>> getPushFormAttachmentRequest(String formId, Path attachment, String token) {
     return RequestBuilder.post(baseUrl)
         .asJsonMap()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/attachments/" + attachment.getFileName().toString())
         .withHeader("Authorization", "Bearer " + token)
         .withHeader("Content-Type", "*/*")
@@ -183,6 +194,7 @@ public class CentralServer implements RemoteServer {
   public Request<Map<String, Object>> getPushSubmissionRequest(String token, String formId, Path submission) {
     return RequestBuilder.post(baseUrl)
         .asJsonMap()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions")
         .withHeader("Authorization", "Bearer " + token)
         .withHeader("Content-Type", "application/xml")
@@ -193,6 +205,7 @@ public class CentralServer implements RemoteServer {
   public Request<Map<String, Object>> getPushSubmissionAttachmentRequest(String token, String formId, String instanceId, Path attachment) {
     return RequestBuilder.post(baseUrl)
         .asJsonMap()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms/" + formId + "/submissions/" + instanceId + "/attachments/" + attachment.getFileName().toString())
         .withHeader("Authorization", "Bearer " + token)
         .withHeader("Content-Type", "*/*")
@@ -203,6 +216,7 @@ public class CentralServer implements RemoteServer {
   public Request<List<RemoteFormDefinition>> getFormsListRequest(String token) {
     return RequestBuilder.get(baseUrl)
         .asJsonList()
+        .withIgnoreCookies()
         .withPath("/v1/projects/" + projectId + "/forms")
         .withHeader("Authorization", "Bearer " + token)
         .withResponseMapper(list -> list.stream()

